app/models/krikri/activity.rb

```diff
@@ -40,18 +39,24 @@ def set_end_time
     ##
     # Runs the block, setting the start and end time of the run. The given block
     # is passed an instance of the agent, and a URI representing this Activity.
+    # 
+    # Handles logging of activity start/stop and failure states.
+    #
+    # @raise [RuntimeError] re-raises logged errors on Activity failure
     def run
       if block_given?
         update_attribute(:end_time, nil) if ended?
+        Krikri::Logger.log :info, "Activity #{id} is running"
```

Comment:
Could this also log the value of `agent.contantize` so that it's easy to tell in the logfile what kind of activity this is?